src/Components/LifecycleTable/lifecycleTable.tsx

```diff
@@ -0,0 +1,129 @@
+  if (activeSortIndex) {
+    sortedRepositories = lifecycleData.sort((a: LifecycleChanges, b: LifecycleChanges) => {
+      const aValue = getSortableRowValues(a)[activeSortIndex];
+      const bValue = getSortableRowValues(b)[activeSortIndex];
+      if (typeof aValue === 'number') {
+        // Numeric sort
+        if (activeSortDirection === 'asc') {
+          return (aValue as number) - (bValue as number);
+        }
+        return (bValue as number) - (aValue as number);
+      } else {
+        // String sort
+        if (activeSortDirection === 'asc') {
+          return (aValue as string).localeCompare(bValue as string);
+        }
+        return (bValue as string).localeCompare(aValue as string);
+      }
+    });
+  }
+
+  const getSortParams = (columnIndex: number): ThProps['sort'] => ({
+    sortBy: {
+      index: activeSortIndex,
+      direction: activeSortDirection,
+      defaultDirection: 'asc' // starting sort direction when first sorting a column. Defaults to 'asc'
+    },
+    onSort: (_event, index, direction) => {
+      setActiveSortIndex(index);
+      setActiveSortDirection(direction);
+    },
+    columnIndex
+  });
+
+  // In this example, we wrap all but the 1st column and make the 1st and 3rd columns sortable just to demonstrate.
+  return (
+    <Table aria-label="Sortable table" ouiaId="SortableTable">
+      <Thead>
+        <Tr>
+          <Th sort={getSortParams(0)}>{columnNames.name}</Th>
+          <Th modifier="wrap" sort={getSortParams(1)}>
+            {columnNames.release}
+          </Th>
+          <Th modifier="wrap" sort={getSortParams(2)}>
+            {columnNames.release_date}
+          </Th>
+          <Th modifier="wrap" sort={getSortParams(3)}>
+            {columnNames.retirement_date}
+          </Th>
+          <Th modifier="wrap" sort={getSortParams(4)}>
+            {columnNames.systems}
+          </Th>
+        </Tr>
+      </Thead>
+      <Tbody>
+        {sortedRepositories.map((repo: { name: string | number | boolean | React.ReactElement<any, string | React.JSXElementConstructor<any>> | Iterable<React.ReactNode> | React.ReactPortal | null | undefined; major: string | number | boolean | React.ReactElement<any, string | React.JSXElementConstructor<any>> | Iterable<React.ReactNode> | React.ReactPortal | null | undefined; minor: string | number | boolean | React.ReactElement<any, string | React.JSXElementConstructor<any>> | Iterable<React.ReactNode> | React.ReactPortal | null | undefined; release: string | number | boolean | React.ReactElement<any, string | React.JSXElementConstructor<any>> | Iterable<React.ReactNode> | React.ReactPortal | null | undefined; release_date: Moment.MomentInput; retirement_date: Moment.MomentInput; systems: string | number | boolean | React.ReactElement<any, string | React.JSXElementConstructor<any>> | Iterable<React.ReactNode> | React.ReactPortal | null | undefined; }, rowIndex: React.Key | null | undefined) => (
```

Comment:
This is really big. Can we take this out and instead define a `type` or `interface` that we then mention here?